packages/aws-cdk-lib/aws-dynamodb/lib/table.ts

```diff
@@ -1788,6 +1798,25 @@ export class Table extends TableBase {
       },
     };
   }
+
+  private renderPointInTimeRecoverySpecification(
+    pointInTimeRecovery?: boolean,
+    recoveryPeriodInDays?: number,
+  ): CfnTable.PointInTimeRecoverySpecificationProperty | undefined {
+    if (pointInTimeRecovery == null && recoveryPeriodInDays === undefined) return undefined;
+
+    let pointInTimeRecoverySpecification: CfnTable.PointInTimeRecoverySpecificationProperty | undefined;
+
+    if (recoveryPeriodInDays !== undefined) {
+      pointInTimeRecoverySpecification = { pointInTimeRecoveryEnabled: true, recoveryPeriodInDays };
+    }
+
+    if (pointInTimeRecovery != null) {
```

Comment:
This was lifted-and-shifted from existing logic, but I think we should refactor this to a more precise check for `undefined`. Thoughts?

```suggestion
    if (pointInTimeRecovery !== undefined) {
```